### PR TITLE
[Easy] Stop input transform tests from being flaky

### PR DESCRIPTION
**Pull request by esantorella**
## Motivation

Tests were flaky.
[x] Break one long test into two
[x] Loosen a tolerance
[x] Set seeds to be randomly chosen from the range (0, 1000), where I made sure this is not flaky

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Ran each test 1000 times with each seed in (0, 1000)

### CODE REVIEW

File: test/models/transforms/test_input.py [hunk 1]
```diff
@@ -259,7 +261,9 @@ def test_normalize(self):
                     [X.min(dim=-2, keepdim=True)[0], X.max(dim=-2, keepdim=True)[0]],
                     dim=-2,
                 )
-                self.assertAllClose(nlz.bounds, expected_bounds)
+                atol = 1e-6 if dtype is torch.float32 else None
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:264]: What do you think of dividing the exponents here by two for double precision, rather than using the defaults implicitly through `None`?
```suggestion
                atol = 1e-6 if dtype is torch.float32 else 1e-12
```

File: test/models/transforms/test_input.py [hunk 2]
```diff
@@ -259,7 +261,9 @@ def test_normalize(self):
...
+                rtol = 1e-4 if dtype is torch.float32 else None
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:265]:
```suggestion
                rtol = 1e-4 if dtype is torch.float32 else 1e-8
```

File: test/models/transforms/test_input.py [hunk 3]
```diff
@@ -591,27 +597,25 @@ def test_chained_input_transform(self):
...
-            round_tf = Round(
-                integer_indices=int_idcs, categorical_features=categorical_feats
-            )
-            self.assertEqual(round_tf.integer_indices.tolist(), int_idcs)
-            self.assertEqual(round_tf.categorical_features, categorical_feats)
-            self.assertTrue(round_tf.training)
-            self.assertFalse(round_tf.approximate)
-            self.assertEqual(round_tf.tau, 1e-3)
-            self.assertTrue(round_tf.equals(Round(**round_tf.get_init_args())))
+    def test_round_transform_init(self):
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:600]: While we're at it ...
```suggestion
    def test_round_transform_init(self) -> None:
```

File: test/models/transforms/test_input.py [hunk 4]
```diff
@@ -757,6 +776,8 @@ def test_round_transform(self):
                 )
 
     def test_log10_transform(self):
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:778]:
```suggestion
    def test_log10_transform(self) -> None:
```

File: test/models/transforms/test_input.py [hunk 5]
```diff
@@ -811,6 +832,8 @@ def test_log10_transform(self):
                 self.assertTrue(torch.equal(log_tf.preprocess_transform(X), X_tf))
 
     def test_warp_transform(self):
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:834]:
```suggestion
    def test_warp_transform(self) -> None:
```

File: test/models/transforms/test_input.py [hunk 6]
```diff
@@ -956,6 +979,8 @@ def test_warp_transform(self):
             self.assertTrue((warp_tf.concentration1 == 3.0).all())
 
     def test_one_hot_to_numeric(self):
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:981]:
```suggestion
    def test_one_hot_to_numeric(self) -> None:
```

File: test/models/transforms/test_input.py [hunk 7]
```diff
@@ -1106,6 +1134,9 @@ def f2(x: Tensor, n_f: int = 1) -> Tensor:
             result = x[..., -2:].unsqueeze(-2)
             return result.expand(*result.shape[:-2], n_f, -1)
 
+        # set seed to range where this is known to not be flaky
+        torch.manual_seed(torch.randint(100, torch.Size([])).item())
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:1138]: Nit: Can add `from random import randint` at the top of the file and then use
```suggestion
        torch.manual_seed(randint(0, 100))
```
which does not allocate an intermediate tensor just to extract an integer.

File: test/models/transforms/test_input.py [hunk 8]
```diff
@@ -1336,6 +1367,9 @@ def test_filter_features(self):
         with self.assertRaises(ValueError):
             FilterFeatures(torch.tensor([0, 1, 1], dtype=torch.long))
 
+        # set seed to range where this is known to not be flaky
+        torch.manual_seed(torch.randint(100, torch.Size([])).item())
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:1371]:
```suggestion
        torch.manual_seed(randint(0, 100))
```

File: test/models/transforms/test_input.py [hunk 9]
```diff
@@ -1042,6 +1067,9 @@ def test_append_features(self):
         with self.assertRaises(ValueError):
             AppendFeatures(torch.ones(3, 4, 2))
 
+        # set seed to range where this is known to not be flaky
+        torch.manual_seed(torch.randint(100, torch.Size([])).item())
```
Comment (SebastianAment) [inline test/models/transforms/test_input.py:1071]: Same as on line 1138 (and there are a couple more).

File: test/models/transforms/test_input.py (hunk 1 above)
Comment (esantorella) [reply test/models/transforms/test_input.py:264]: makes sense -- the defaults are `rtol=1e-5`, `atol=1e-8`, so that change is stricter
